caesars-palace/blocks/carousel/carousel.js

```diff
@@ -743,9 +692,20 @@ export default async function decorate(block) {
     if (entries.some((e) => e.isIntersecting)) {
       handleAutoScroll(entries);
     }
-  }, intersectionOptions);
+  }, {
+    root: null,
+    rootMargin: '0px',
+    threshold: 1.0,
+  });
   observer.observe(block);
 
+  mediaVideoWidthQueryMatcher.addEventListener('change', mediaVideoWidthChangeHandler);
+  mediaTextWidthQueryMatcher.addEventListener('change', mediaTextWidthChangeHandler);
+  mediaExtraLargeWidthQueryMatcher.addEventListener('change', mediaExtraLargeWidthChangeHandler);
+  mediaLargeWidthQueryMatcher.addEventListener('change', mediaLargeWidthChangeHandler);
+  mediaMediumWidthQueryMatcher.addEventListener('change', mediaMediumWidthChangeHandler);
+  mediaSmallWidthQueryMatcher.addEventListener('change', mediaSmallWidthChangeHandler);
```

Comment:
Is there any duplicated code in all these handlers? You could have one handler that gets called for changes to multiple media matchers. THen inside the handler instead of relying on `event` you could just ask the matcher directly.  eg. `mediaVideoWidthQueryMatcher.matches`. If all the handlers are different then there isn't much advantage of sharing handlers however.

Reply:
`mediaVideoWidthQueryMatcher` and `mediaTextWidthQueryMatcher` are very different from the others. 
The others have similar code but different parameters in the function calls they use.